Automattic/wp-calypso
# P2 revised signup flow: add email confirmation step

#### Changes proposed in this Pull Request

- Adds an email confirmation step to the revised P2 flow, as described in p9lV3a-2Lp-p2#comment-3599. This step comes before the P2 workspace creation step.
- Removes the `p2-get-started` step from the flow.

#### Testing instructions



* Apply D71249-code and D71110-code.
* Use `ENABLE_FEATURES=no-force-sympathy yarn start` when building, to disable a dev-only feature that randomly clears out persistent browser state.

1. Visit `calypso.localhost:3000/start/p2-new` in incognito.
2. You should see the P2 account creation screen. Please sign up with a test email that you have access to, as this will be needed later.
3. Upon account creation, you should get the email confirmation screen next, i.e. "Check your email" screen.
4. Open your inbox in the same browser and verify that an activation email was sent, i.e. "Activate <username>". Do not activate anything yet.
5. Go back to the signup flow tab, and verify that the 'Resend email' feature works.
6. Go back to the email, and click the "Confirm" button. Notice that a new browser tab opens and redirects to `wordpress.com/start/p2-new/p2-get-started`. This is okay. If you change `wordpress.com` to `calypso.localhost:3000`, you should get the P2 workspace creation site as expected.

Some other things to test:
* Unverified accounts should always get this screen. Verified accounts should not.


Related to 4166-gh-Automattic/p2

#### Screens
<img width="1680" alt="Screen Shot 2021-12-09 at 6 51 45 PM" src="https://user-images.githubusercontent.com/730823/145383387-6a0779bb-7a54-417d-b3cd-ba654e9a5216.png">
